CommonMark/CommonMark.py

```diff
@@ -49,15 +49,13 @@ def prepare(block):
         block.__dict__[r] = None
     if block.is_open is not None:
         block.__dict__['open'] = block.is_open
-        del(block.is_open)
+        del block.is_open
+
     # trim empty elements...
-    for attr in dir(block):
-        if not callable(attr) and not attr.startswith("__") and \
-           attr not in ['pretty', 'is_container',
-                        'append_child', 'prepend_child', 'unlink',
-                        'insert_after', 'insert_before', 'walker']:
-            if block.__dict__[attr] in ["", [], None, {}]:
-                del(block.__dict__[attr])
+    for attr, value in list(block.__dict__.items()):
```

Comment:
I think it's ok provided it's done on instances not classes. Either way, the existing code is terrible, since it mutilates any object passed into it under the pretense of jsonifying it. But all this commit aims to do is make it output the format it claims to, rather than fix a super-janky implementation